modules/xml-sitemap/sitemap.php

```diff
@@ -0,0 +1,595 @@
+	unset( $tree );
+	if ( ! empty( $xml ) ) {
+		set_transient( 'jetpack_sitemap', $xml, DAY_IN_SECONDS );
+		echo $xml;
+	}
+
+	die();
+}
+
+/**
+ * Prints the news XML sitemap conforming to the Sitemaps.org protocol.
+ * Outputs an XML list of up to 1000 posts published in the last 2 days.
+ *
+ * @link http://sitemaps.org/protocol.php Sitemaps.org protocol
+ */
+function jetpack_print_news_sitemap() {
+
+	$xml = get_transient( 'jetpack_news_sitemap' );
+
+	if ( $xml ) {
+		header( 'Content-Type: application/xml' );
+		echo $xml;
+		die();
+	}
+
+	global $wpdb;
+
+	/**
+	 * Filter post types to be included in news sitemap.
+	 *
+	 * @since 3.9
+	 *
+	 * @param array $post_types Array with post types to include in news sitemap.
+	 */
+	$post_types = apply_filters( 'jetpack_sitemap_news_sitemap_post_types', array ( 'post' ) );
+	if ( empty( $post_types ) ) {
+		return;
+	}
+
+	$post_types_in = array ();
+	foreach ( $post_types as $post_type ) {
+		$post_types_in[] = $wpdb->prepare( '%s', $post_type );
+	}
+	$post_types_in_string = implode( ', ', $post_types_in );
+
+	/**
+	 * Filter limit of entries to include in news sitemap.
+	 *
+	 * @since 3.9
+	 *
+	 * @param int $count Number of entries to include in news sitemap.
+	 */
+	$limit        = apply_filters( 'jetpack_sitemap_news_sitemap_count', 1000 );
+	$cur_datetime = current_time( 'mysql', true );
+
+	$query = $wpdb->prepare( "
+		SELECT p.ID, p.post_title, p.post_type, p.post_date, p.post_name, p.post_date_gmt, GROUP_CONCAT(t.name SEPARATOR ', ') AS keywords
+		FROM
+			$wpdb->posts AS p LEFT JOIN $wpdb->term_relationships AS r ON p.ID = r.object_id
+			LEFT JOIN $wpdb->term_taxonomy AS tt ON r.term_taxonomy_id = tt.term_taxonomy_id AND tt.taxonomy = 'post_tag'
+			LEFT JOIN $wpdb->terms AS t ON tt.term_id = t.term_id
+		WHERE
+			post_status='publish' AND post_type IN ( {$post_types_in_string} ) AND post_date_gmt > (%s - INTERVAL 2 DAY)
+		GROUP BY p.ID
+		ORDER BY p.post_date_gmt DESC LIMIT %d", $cur_datetime, $limit );
+
+	header( 'Content-Type: application/xml' );
+	ob_start();
+	echo '<?xml version="1.0" encoding="UTF-8"?>' . "\n";
+	?>
+	<!-- generator="jetpack" -->
+	<urlset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	        xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd"
+	        xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
+	        xmlns:news="http://www.google.com/schemas/sitemap-news/0.9"
+	        xmlns:image="http://www.google.com/schemas/sitemap-image/1.1"
+		>
+		<?php
+		$posts = $wpdb->get_results( $query );
+		foreach ( $posts as $post ):
+
+			/**
+			 * Filter condition to allow skipping specific posts in news sitemap.
+			 *
+			 * @since 3.9
+			 *
+			 * @param bool $skip Current boolean. False by default, so no post is skipped.
+			 * @param WP_POST $post Current post object.
+			 */
+			if ( apply_filters( 'jetpack_sitemap_news_skip_post', false, $post ) ) {
```

Comment:
I would lean towards agreeing with @eliorivero's point of view here-- it's a very easy thing from the user's perspective to accommodate both filters, but if they had a use case for only wanting to affect one sitemap or the other, it would be very difficult without having these as separate filters.